service/history/replicationTaskFetcher.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package history
+
+import (
+	"context"
+	"time"
+
+	"github.com/uber/cadence/.gen/go/cadence/workflowserviceclient"
+	r "github.com/uber/cadence/.gen/go/replicator"
+	"github.com/uber/cadence/common/backoff"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/log/tag"
+)
+
+const (
+	timerFireInterval  = 2 * time.Second
+	timerRetryInterval = 1 * time.Second
+	timerJitter        = 0.15
+)
+
+type replicationTaskFetcher struct {
+	sourceCluster string
+	numFetchers   int
+	logger        log.Logger
+	remotePeer    workflowserviceclient.Interface
+	requestChan   chan *request
+	done          chan struct{}
+}
+
+func newReplicationTaskFetcher(logger log.Logger, sourceCluster string, numFetchers int, sourceFrontend workflowserviceclient.Interface) *replicationTaskFetcher {
+	return &replicationTaskFetcher{
+		numFetchers:   numFetchers,
+		logger:        logger,
+		remotePeer:    sourceFrontend,
+		sourceCluster: sourceCluster,
+		requestChan:   make(chan *request, 1000),
```

Comment:
that 1000 should be defined as constant at the beginning of the file

Reply:
done